tests/http/clients/asgi.py

```diff
@@ -162,13 +162,12 @@ async def ws_connect(
         *,
         protocols: list[str],
     ) -> AsyncGenerator[WebSocketClient, None]:
-        try:
-            with self.client.websocket_connect(url, protocols) as ws:
-                yield AsgiWebSocketClient(ws)
-        except WebSocketDisconnect as error:
-            ws = AsgiWebSocketClient(None)
-            ws.handle_disconnect(error)
-            yield ws
+        with self.client.websocket_connect(url, protocols) as ws:
+            ws_client = AsgiWebSocketClient(ws)
+            try:
+                yield ws_client
+            except WebSocketDisconnect as error:
+                ws_client.handle_disconnect(error)
```

Comment:
this is not functionally equivalent, can Disconnect not be thrown during the context manager init?
Also, the except part doesn't seem to be covered by tests, maybe we should change that

Reply:
The change is intentional: the init method can not throw.

Thanks for pointing out the test coverage. Before this change, the except block was invalid, telling us that we don't actually have tests relying on it. Since this is a client used solely for testing, we should probably just remove it instead of adding an artificial test just for coverage of code only used for testing.

Reply:
Fine with that, let's add a comment detailing this though.

Reply:
It turned out the special cleanup code is a relic from before our HTTP & WS tests and test clients were shared. I have fully removed it now.